openmrs-19/src/test/java/org/motechproject/openmrs19/tasks/OpenMRSTaskDataProviderTest.java

```diff
@@ -47,6 +52,57 @@ public void shouldReturnNullWhenClassIsNotSupported() {
     }
 
     @Test
+    public void shouldReturnNullWhenWrongLookupNameForEncounter() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Object object = taskDataProvider.lookup(className, "wrongLookupName", null);
+
+        assertNull(object);
+        verifyZeroInteractions(encounterService);
+    }
+
+    @Test
+    public void shouldReturnNullWhenEmptyLookupFieldsForLookupGetEncounterByUuid() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Map<String, String> lookupFields = new HashMap<>();
+
+        when(encounterService.getEncounterByUuid(null)).thenReturn(null);
+        Object object = taskDataProvider.lookup(className, BY_UUID, lookupFields);
+
+        assertNull(object);
+        verify(encounterService).getEncounterByUuid(null);
+    }
+
+    @Test
+    public void shouldReturnNullWhenEncounterNotFoundForLookupGetEncounterByUuid() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Map<String, String> lookupFields = new HashMap<>();
+        lookupFields.put(UUID, "4");
+
+        when(encounterService.getEncounterByUuid("4")).thenReturn(null);
+        Object object = taskDataProvider.lookup(className, BY_UUID, lookupFields);
+
+        assertNull(object);
+        verify(encounterService).getEncounterByUuid("4");
+    }
+
+    @Test
+    public void shouldReturnEncounterForLookupGetEncounterByUuid() {
```

Comment:
The most important thing in this test is assert checking if object is null and verification if getEncounterByUuid method was invoked with parameter value "4".

when().thenReturn() is faking behaviour of encounterService. A user is trying to retrieve Encounter with Uuid "4" and we want to return null, which means that Encounter is not found. However in the test we can define many when().thenReturn(), so at the end of the test we make sure that getEncounterByUuid was invoked with "4"